src/jit/codegenarmarch.cpp

```diff
@@ -962,6 +962,14 @@ void CodeGen::genPutArgSplit(GenTreePutArgSplit* treeNode)
         // Generate code to load the address that we need into a register
         genConsumeAddress(addrNode);
         addrReg = addrNode->gtRegNum;
+
+        // If addrReg equal to baseReg, we use the last target register as alternative baseReg.
+        // Because baseReg is not allocated as one of target register by masking,
+        // we can avoid baseReg, addrReg, and the last target register are all same.
```

Comment:
Actually, on further thought, I don't believe that `addrReg` should ever be equal to `baseReg`, because the liveness model for LSRA ensures that an internal register doesn't conflict with sources. (See the comment at the beginning of lsra.cpp). So there should be an assert here, and if it fires it indicates a bug.